azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeSecurityGroupExtension.java

```diff
@@ -123,6 +123,9 @@
       ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(regionAndId.region());
 
       VirtualMachine vm = api.getVirtualMachineApi(resourceGroup.name()).get(regionAndId.id());
```

Comment:
I'd say it is ok as-is? taht check would throw a NPE, but in this case an IAE makes more sense?